File: kompost-proxy/src/main/java/ElmFiles.java
import okhttp3.*;
import org.apache.camel.Header;

import java.io.IOException;

import static org.apache.camel.Exchange.HTTP_PATH;

public class ElmFiles {
    OkHttpClient client = new OkHttpClient();
    final String targetDir;

    public ElmFiles(String targetDir) {
        this.targetDir = targetDir;
    }

    public String perform(@Header(HTTP_PATH) String theFile
    ) throws IOException {
        Request request = new Request.Builder()
                .url("http://localhost:8000/" + targetDir + "/" + theFile)
                .build();
        Response response = null;
        try {
            response = client.newCall(request).execute();
            String result = response.body().string();
            return result;
        } finally {
            response.body().close();
            response.close();
        }
    }
}

File: kompost-proxy/src/main/java/ElmFiles.kt
import okhttp3.*
import org.apache.camel.Header
import java.io.IOException
import org.apache.camel.Exchange.HTTP_PATH

class ElmFiles(internal val targetDir: String) {
    internal var client = OkHttpClient()

    fun perform(@Header(HTTP_PATH) filename: String): String {
        val request = Request.Builder()
                .url("http://localhost:8000/$targetDir/$filename")
                .build()
        client.newCall(request).execute().use {response ->
            return if (response.isSuccessful) {
                response.body()!!.string()
            } else {
                throw IOException("Could perform $filename. ${response.networkResponse()!!.message()}")
            }
        }
    }
}

